lib/protractor.js

```diff
@@ -414,6 +448,48 @@ Protractor.prototype.addBaseMockModules_ = function() {
           if ($compileProvider.debugInfoEnabled) {
             $compileProvider.debugInfoEnabled(true);
           }
+        }]).
+        config(['$provide', function($provide) {
+          $provide.decorator('$timeout', ['$delegate', function($delegate) {
+            var timeout = $delegate;
+
+            var taskId = 0;
+            if (!window.pendingTimeouts) {
```

Comment:
Let's make this more debuggey and less likely to collide, what about `window.NG_PENDING_TIMEOUTS`